sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/group/ThroughputGroupControllerBase.java

```diff
@@ -158,6 +158,7 @@ public <T> Mono<T> processRequest(RxDocumentServiceRequest request, Mono<T> orig
         return this.resolveRequestController()
             .flatMap(requestController -> {
                 if (requestController.canHandleRequest(request)) {
+                    request.setPriorityLevel(group.getPriorityLevel());
```

Comment:
I would probably create a whole separate throughput group controller for the priority throttling. Because all SDK need to do is set the header and pass throughput.

for the existing throughput group controller, it has to do a lot more processing - like read container throughput, create request handler, partition handler etc. All of these are not need for priority throttling 

we can have two major categories for throughput group controller:
ClientBasedThroughputGroupController
ServerBasedThroughputGroupController

Reply:
I disagree - priority IMO should be an additional config - it could become useful to combine throughput threshold with priority.

Reply:
So, Bala's approach looks good to me